flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EventHelper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.utils;
+
+import org.apache.flink.kubernetes.operator.crd.AbstractFlinkResource;
+import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
+import org.apache.flink.kubernetes.operator.listener.FlinkResourceListener;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.function.BiConsumer;
+
+/** Helper class for status management and updates. */
+public class EventHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventHelper.class);
+
+    private final KubernetesClient client;
+    private final BiConsumer<AbstractFlinkResource<?, ?>, Event> eventListener;
+
+    public EventHelper(
+            KubernetesClient client, BiConsumer<AbstractFlinkResource<?, ?>, Event> eventListener) {
+        this.client = client;
+        this.eventListener = eventListener;
+    }
+
+    public boolean triggerEvent(
+            AbstractFlinkResource<?, ?> resource,
+            EventUtils.Type type,
+            String reason,
+            String message,
+            EventUtils.Component component) {
+        return EventUtils.createOrUpdateEvent(
```

Comment:
It would also be possible to handle the k8s event create/update as listener implementation, but perhaps that is overkill when the intention is to keep it hard wired?

Reply:
I feel that would be slightly weird because the listener is actually listening to the events :) Technically it could be done to create the event and add a hardwired listener that sends it to kubernetes but I don't think it would make the logic easier to understand.